source/state_representation/include/state_representation/Space/Cartesian/CartesianTwist.hpp

```diff
@@ -5,247 +5,295 @@
+   */
+  CartesianTwist(const CartesianPose& pose);
+
+  /**
+   * @brief Construct a CartesianTwist from a linear_velocity given as a vector of coordinates for the linear velocity.
+   */
+  explicit CartesianTwist(const std::string& name, const Eigen::Vector3d& linear_velocity, const std::string& reference = "world");
+
+  /**
+   * @brief Construct a CartesianTwist from a linear_velocity given as a vector of coordinates and a quaternion.
+   */
+  explicit CartesianTwist(const std::string& name, const Eigen::Vector3d& linear_velocity, const Eigen::Vector3d& angular_velocity, const std::string& reference = "world");
+
+  /**
+   * @brief Construct a CartesianTwist from a single 6d twist vector
+   */
+  explicit CartesianTwist(const std::string& name, const Eigen::Matrix<double, 6, 1>& twist, const std::string& reference = "world");
+
+  /**
+   * @brief Constructor for the zero twist
+   * @param name the name of the state
+   * @param the name of the reference frame
+   * @return CartesianTwist with zero values
+   */
+  static const CartesianTwist Zero(const std::string& name, const std::string& reference = "world");
+
+  /**
+   * @brief Constructor for a random twist
+   * @param name the name of the state
+   * @param the name of the reference frame
+   * @return CartesianTwist random twist
+   */
+  static const CartesianTwist Random(const std::string& name, const std::string& reference = "world");
+
+  /**
+   * @brief Copy assignement operator that have to be defined to the custom assignement operator
+   * @param twist the twist with value to assign
+   * @return reference to the current twist with new values
+   */
+  CartesianTwist& operator=(const CartesianTwist& twist);
+
+  /**
+   * @brief Overload the = operator from a CartesianState
+   * @param state CartesianState to get velocity from
+   */
+  CartesianTwist& operator=(const CartesianState& state);
+
+  /**
+   * @brief Set the values of the 6D twist from a 6D Eigen Vector
+   * @param twist the twist as an Eigen Vector
+   */
+  CartesianTwist& operator=(const Eigen::Matrix<double, 6, 1>& twist);
+
+  /**
+   * @brief Overload the *= operator
+   * @param twist CartesianTwist to multiply with
+   * @return the current CartesianTwist multiplied by the CartesianTwist given in argument
+   */
+  CartesianTwist& operator*=(const CartesianTwist& twist);
+
+  /**
+   * @brief Overload the * operator with a twist
+   * @param twist CartesianTwist to multiply with
+   * @return the current CartesianTwist multiplied by the CartesianTwist given in argument
+   */
+  const CartesianTwist operator*(const CartesianTwist& twist) const;
+
+  /**
+   * @brief Overload the * operator with a state
+   * @param state CartesianState to multiply with
+   * @return the current CartesianTwist multiplied by the CartesianState given in argument
+   */
+  const CartesianState operator*(const CartesianState& state) const;
```

Comment:
When returned a state = twist * state, is that essentially an integration? If so is the unit timestep just implied?

Reply:
This is a good catch both operator (here and on wrench) should not be here anymore. Those are relicates when operators where defined at each class level not on the main CartesianState.